src/main/java/org/mule/module/apikit/metadata/internal/amf/MetadataFactory.java

```diff
@@ -43,11 +46,11 @@ public static MetadataType fromJSONSchema(final Shape shape, String example) {
     if (shape instanceof AnyShape) {
       final AnyShape anyShape = (AnyShape) shape;
 
-      final TypeLoader typeLoader = anyShape.isDefaultEmpty() && !example.isEmpty()
-          ? createJsonExampleTypeLoader(example)
-          : new JsonTypeLoader(anyShape.buildJsonSchema());
-
+      TypeLoader typeLoader = new JsonTypeLoader(APIConfiguration.API().elementClient().buildJsonSchema(anyShape));
       metadataType = typeLoader.load(null);
+      if ((!metadataType.isPresent() || metadataType.get() instanceof AnyType) && !example.isEmpty()) {
+        metadataType = createJsonExampleTypeLoader(example).load(null);
+      }
```

Comment:
Just to keep in mind, this is a change that I made to keep the impact of the upgrade as seamless as possible but in fact this is changing the order of priority in which we consider the metadata. 
AMF 5 has good improvements on example resolution so we can consider keeping the original order and updating the tests to take advantage of the examples (I'll elaborate a new PR with this to b be more clear)